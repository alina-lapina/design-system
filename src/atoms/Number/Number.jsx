import React from 'react';
import PropTypes from 'prop-types';

const formatNumber = num => num.toString()
	.replace(/\D+/g, '') // remove everything that is not a number
	.replace(/(\d)(?=(\d{3})+(?!\d))/g, ' '); // add spaces in desired format

const Number = ({
	size, number,
}) => (
	<div className={`number-component ${size}`}>
		{formatNumber(number)}
	</div>
);

Number.defaultProps = {
	size: 'small',
};

Number.propTypes = {
	size: PropTypes.oneOf(['small', 'medium', 'large']),
	number: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
	]).isRequired,
};

export default Number;
