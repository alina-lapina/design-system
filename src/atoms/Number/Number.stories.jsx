import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Number from './Number';

storiesOf('Number', module).addDecorator(centered)
	.add('Small', () => (
		<div>
			<Number number="599246391" size="small" />
		</div>
	))
	.add('Medium', () => (
		<div>
			<Number number="59923" size="medium" />
		</div>
	))
	.add('Large', () => (
		<div>
			<Number number="789398" size="large" />
		</div>
	));
