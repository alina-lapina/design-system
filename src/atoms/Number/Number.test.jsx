import React from 'react';
import { shallow } from 'enzyme';
import Number from './Number';

describe('Number component', () => {
	test('Matches the snapshot', () => {
		const numberComponent = shallow(<Number number="23423" />);
		expect(numberComponent).toMatchSnapshot();
	});
});
