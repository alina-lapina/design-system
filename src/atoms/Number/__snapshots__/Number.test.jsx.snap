// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number component Matches the snapshot 1`] = `
<div
  className="number-component small"
>
  2 423
</div>
`;
